dynamic/js/basho/code-blocks.coffee

```diff
@@ -0,0 +1,462 @@
+###
+Code Blocks; Tabs and Titles, Generation and Interaction
+========================================================
+
+When a <pre><code> block is created using a fenced code block that's been given
+an explicit language we will want to pass it though Highlight.js to give us
+something to colorize.
+
+
+When such a code block appears on its own, we will want to prepend a title to
+that element to show what kind of file is being previewed.
+
+For reference, Markdown will generate structures similar to the below,
+
+  <pre><code class="language-python">
+    . . .
+  </code></pre>
+
+We aim to build that into,
+
+  <div class="code-block--titled">
+    <span class="inline-block code-block__title">
+      Python
+    </span>
+    <pre class="code-block__code"><code class="language-python">
+      . . .
+    </code></pre>
+  </div>
+
+
+When multiple <pre><code> blocks appear next to one another, we want to collapse
+the separate blocks into a single element with languages selectable via tabs.
+
+For reference, Markdown will generate structures similar to the below,
+
+  <pre><code class="language-ruby">
+    . . .
+  </code></pre>
+  <pre><code class="language-java">
+    . . .
+  </code></pre>
+
+We aim to build that into,
+
+  <div class="code-block--tabbed">
+    <div class="code-block__tab-set-wrapper">
+      <div class="float-left    code-block__edge-fader--left "><span class="inline-block   code-block__edge-fader__arrow"></span></div>
+      <div class="float-right   code-block__edge-fader--right"><span class="inline-block   code-block__edge-fader__arrow"></span></div>
+      <ul class="overflow-x   code-block__tab-set">
+        <li class="inline-block   code-block__tab code-block__tab--active">
+          <a class="block" href="#code-block__language-java000" data-language="language-java">
+            Java
+          </a>
+        </li>
+        <li class="inline-block   code-block__tab">
+          <a class="block" href="#code-block__language-python000" data-language="language-python">
+            Python
+          </a>
+        </li>
+      </ul>
+    </div>
+    <div class="code-block__code-set">
+      <pre class="code-block__code" id="code-block__java000"><code class="language-java">
+        . . .
+      </code></pre>
+      <pre class="code-block__code" id="code-block__python000"><code class="language-python hljs">
+        . . .
+      </code></pre>
+    </div>
+  </div>
+###
+
+#TODO: strict mode enables a pretty large number of runtime checks. We probably
+#      want to turn it off when we deploy to prod.
+#      Perhaps we can figure out some way to enable it when building debug?
+'use strict'
```

Comment:
FWIW, strict mode is always on in the Riak node.js client. I doubt there would be any noticable performance change in the browser, too.